rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -106,7 +106,10 @@ void SequentialReader::open(
     if (!storage_) {
       throw std::runtime_error{"No storage could be initialized from the inputs."};
     }
-    storage_->set_read_order(read_order_);
+    if (!set_read_order(read_order_)) {
+      ROSBAG2_CPP_LOG_WARN(
+        "Could not set read order on open(), defaulting to storage plugin's default read order");
+    }
```

Comment:
@james-rms @emersonknapp I believe we shall throw exception here.
We can't fall back on the storage plugin's default read order since we don't know what is default for it.
We need to be assure that `read_order_` in `SequantialReader` corresponds to the read order in storage plugin.
If we can't be assure - this is undefined behavior and we shall not proceed - in this case.